dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityPreviewServlet.java

```diff
@@ -0,0 +1,359 @@
+package com.dotcms.rendering.velocity.servlet;
+
+import com.dotcms.business.CloseDBIfOpened;
+import com.dotcms.enterprise.LicenseUtil;
+import com.dotcms.enterprise.license.LicenseLevel;
+import com.dotcms.rendering.velocity.services.PageContextBuilder;
+import com.dotcms.rendering.velocity.services.VelocityType;
+import com.dotcms.rendering.velocity.util.VelocityUtil;
+import com.dotcms.rendering.velocity.viewtools.DotTemplateTool;
+import com.dotcms.rendering.velocity.viewtools.RequestWrapper;
+import com.dotcms.rendering.velocity.viewtools.content.ContentMap;
+import com.dotcms.visitor.business.VisitorAPI;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.web.HostWebAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.filters.CMSFilter;
+import com.dotmarketing.filters.Constants;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.model.IHTMLPage;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.UtilMethods;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.URLDecoder;
+import java.util.Map;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.velocity.Template;
+import org.apache.velocity.context.Context;
+import org.apache.velocity.exception.MethodInvocationException;
+import org.apache.velocity.exception.ParseErrorException;
+import org.apache.velocity.exception.ResourceNotFoundException;
+
+import com.liferay.portal.language.LanguageException;
+import com.liferay.portal.language.LanguageUtil;
+import com.liferay.portal.model.User;
+
+public class VelocityPreviewServlet extends HttpServlet {
+
+    private static final long serialVersionUID = 1L;
+
+
+    private static PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+
+
+
+    private static HostWebAPI hostWebAPI = WebAPILocator.getHostWebAPI();
+
+    private static VisitorAPI visitorAPI = APILocator.getVisitorAPI();
+
+    public static void setPermissionAPI(PermissionAPI permissionAPIRef) {
+        permissionAPI = permissionAPIRef;
+    }
+
+    private String CHARSET = "UTF-8";
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this misleading mutable servlet instance field or make it "static" and/or "final" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2226)